cmake/clang_tidy.cmake

```diff
@@ -11,7 +11,7 @@ find_program(RUN_CLANG_TIDY NAMES run-clang-tidy run-clang-tidy-13 run-clang-tid
 message(STATUS "RUN_CLANG_TIDY=${RUN_CLANG_TIDY}")
 if (RUN_CLANG_TIDY)
     # disable all compiler warnings since we are just interested in the tidy ones
-    add_custom_target(run-clang-tidy ${RUN_CLANG_TIDY} -checks=-performance-unnecessary-copy-initialization -p=${CMAKE_BINARY_DIR} -j ${NPROC} -extra-arg=-w -quiet)
+    add_custom_target(run-clang-tidy ${RUN_CLANG_TIDY} -p=${CMAKE_BINARY_DIR} -j ${NPROC} -extra-arg=-w -quiet)
```

Comment:
@danmar This would have affect all source files.
Interestingly this is also something we check for so the self-check should have detected this as well.

Reply:
It's also possible the `-checks` option might override the existing `.clang-tidy` configuration - which would be a bad thing.

Reply:
> It's also possible the `-checks` option might override the existing `.clang-tidy` configuration - which would be a bad thing.

Actually it is appended:
```
--checks=<string>              -
[...]
                                   This option's value is appended to the
                                   value of the 'Checks' option in .clang-tidy
                                   file, if any.
```